### PR TITLE
Update to support multiple key types.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
wip, started by @JSAssassin.

### CODE REVIEW

**Comment by mattcollier** [comment]
@JSAssassin please update broken dev dep branches so that we can hopefully see CI passing.

**Comment by dmitrizagidulin** [comment]
@mattcollier the code is still on the operating table; tests won't be passing until we refactor some stuff.

File: KmsClient.js [hunk 1]
```diff
@@ -4,9 +4,10 @@
 import base64url from 'base64url-universal';
 import {httpClient, DEFAULT_HEADERS} from '@digitalbazaar/http-client';
 import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
-import {SECURITY_CONTEXT_V2_URL} from 'jsonld-signatures';
-
+import * as sec from 'security-context';
 import {cryptoLd} from './cryptoLd.js';
+const securityContext = sec.constants.SECURITY_CONTEXT_V2_URL;
```
Comment (mattcollier) [inline KmsClient.js:9]: If not using SECURITY_CONTEXT_V2_URL, I'd still like to see some all cap var name that has the term `URL` in it to specify that it's a URL and not a context document or other.

**Review by mattcollier — APPROVED** [review]
One comment.  TY.

File: KeystoreAgent.js [hunk 2]
```diff
@@ -68,8 +71,12 @@ export class KeystoreAgent {
     } else if(type === 'Ed25519VerificationKey2018') {
       type = 'Ed25519VerificationKey2018';
       Class = AsymmetricKey;
-    } else if(type === 'keyAgreement' || type === 'X25519KeyAgreementKey2019') {
-      type = 'X25519KeyAgreementKey2019';
+    } else if(type === 'Ed25519VerificationKey2020') {
+      type = 'Ed25519VerificationKey2020';
+      Class = AsymmetricKey;
+    } else if(type === 'keyAgreement' || type === 'X25519KeyAgreementKey2019' ||
+              type === 'X25519KeyAgreementKey2020') {
```
Comment (dlongley) [inline KeystoreAgent.js:78]:
```suggestion
    } else if(type === 'keyAgreement' || type === 'X25519KeyAgreementKey2019' ||
      type === 'X25519KeyAgreementKey2020') {
```

File: KmsClient.js [hunk 3]
```diff
@@ -4,8 +4,9 @@
 import base64url from 'base64url-universal';
 import {httpClient, DEFAULT_HEADERS} from '@digitalbazaar/http-client';
 import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
-
-const SECURITY_CONTEXT_V2_URL = 'https://w3id.org/security/v2';
+import * as sec from 'security-context';
+import {cryptoLd} from './cryptoLd.js';
+const securityContext = sec.constants.SECURITY_CONTEXT_V2_URL;
```
Comment (dlongley) [inline KmsClient.js:9]:
```suggestion
const securityContextUrl = sec.constants.SECURITY_CONTEXT_V2_URL;
```

**Review by dlongley — APPROVED** [review]
AFAICT, LGTM. Please note @mattcollier's comment -- we should rename that variable.

File: KeystoreAgent.js (hunk 2 above)
Comment (JSAssassin) [reply KeystoreAgent.js]: Updated here ddb283e

File: KmsClient.js (hunk 3 above)
Comment (JSAssassin) [reply KmsClient.js]: Updated here ddb283e

**Review by mattcollier — APPROVED** [review]
TY. Merging and releasing now.

**Pull request merged**